### DISCUSSION TITLE
Missing semicolon in Rust statements causes some failures

### DISCUSSION THREAD

**Discussion by mmdbalkhi** [sigmaSd/IRust#136] [General]
I encountered an issue with IRust. Specifically, when I initially entered `println!("Hello World!");`, the output was displayed correctly. However, when I input the second part of the program, the output of this part did not display as expected. I hope the examples illustrate the failures clearly.

```rust
In: println!("Hello World!");
Out: Hello World!

In: println!("Hello World!")
Out: Hello World!
Hello World!
()

In: print!("Hiii!");
In: 
In: print!("Hiii!")
Out: Hello World!
Hiii!Hiii!()
```
In other words, the previous commands are repeated.

* `/tmp/irust_repl/.../src/main.rs`:
```rust
fn main() -> (){
      #[allow(unused_imports)]use irust_prelude::*;
      println!("Hello World!");
      print!("Hiii!");
      #[allow(unreachable_code)]() // Do not write past this line (it will corrupt the repl)
}
```

* `/tmp/irust_repl/.../src/main_extern.rs`:
```rust
fn main() -> (){
  #[allow(unused_imports)]use irust_prelude::*;
  println!("Hello World!");
  print!("Hiii!");
  #[allow(unreachable_code)]() // Do not write past this line (it will corrupt the repl)
}
```

**Comment by sigmaSd** [comment]
Hello, it seems to me its working as expected

- If a line ends with semicolon its considered a statement and its not executed ,
- if it doesn't its considered an expression and its executed without being saved in the repl

- so `println!(..);` you will not see the result but it will get inserted in the repl, then if you evaluate anything , you'll always see that expression
- `println!()` will just print the content immediately without inserting in the repl, so if the next expressions you evaluate wont show it again

Also I think using `prints` is not really needed in a repl, if you want to see something you can just execute it 

![image](https://github.com/sigmaSd/IRust/assets/22427111/9a0f765a-063c-4f7c-b723-f7539cf2e346)
